process.go

```diff
@@ -0,0 +1,263 @@
+	title := "HCSShim::Process::ResizeConsole"
+	var (
+		resultp *uint16
+	)
+
+	modifyRequest := newModifyConsoleRequest(width, height)
+
+	logrus.Debugf("Setting console size to: %d, %d, %s", width, height, modifyRequest)
+
+	modifyRequestb, err := json.Marshal(modifyRequest)
+	if err != nil {
+		return err
+	}
+
+	modifyRequestStr := string(modifyRequestb)
+
+	if err := hcsModifyProcess(process.handle, modifyRequestStr, &resultp); err != nil {
+		err = makeError(err, title, modifyRequestStr)
+		logrus.Error(err)
+		return err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s : %s", title, result, modifyRequestStr)
+	}
+
+	return nil
+}
+
+// Stdio returns the stdin, stdout, and stderr pipes, respectively. Closing
+// these pipes does not close the underlying pipes; it should be possible to
+// call this multiple times to get multiple interfaces.
+func (process *hcsProcess) Stdio() (io.WriteCloser, io.ReadCloser, io.ReadCloser, error) {
+	title := "HCSShim::Process::Stdio"
+	logrus.Debug(title)
+	var (
+		processInfo HcsProcessInformation
+		resultp     *uint16
+	)
+
+	if err := hcsGetProcessInfo(process.handle, &processInfo, &resultp); err != nil {
+		err = makeError(err, title, "")
+		logrus.Error(err)
+		return nil, nil, nil, err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s", title, result)
+	}
+
+	logrus.Debugf("Returned processInfo %s", processInfo)
+
+	pipes, err := makeOpenFiles([]syscall.Handle{processInfo.StdInput, processInfo.StdOutput, processInfo.StdError})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return pipes[0], pipes[1], pipes[2], nil
+}
+
+// CloseStdin closes the write side of the stdin pipe so that the process is
+// notified on the read side that there is no more data in stdin.
+func (process *hcsProcess) CloseStdin() error {
+	title := "HCSShim::Process::CloseStdin"
+	var (
+		resultp *uint16
+	)
+
+	modifyRequest := newCloseHandleRequest(stdIn)
+
+	modifyRequestb, err := json.Marshal(modifyRequest)
+	if err != nil {
+		return err
+	}
+
+	modifyRequestStr := string(modifyRequestb)
+
+	if err := hcsModifyProcess(process.handle, modifyRequestStr, &resultp); err != nil {
+		err = makeError(err, title, "")
+		logrus.Error(err)
+		return err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s", title, result)
+	}
+
+	return nil
+}
+
+// Close cleans up any state associated with the process but does not kill
+// or wait on it.
+func (process *hcsProcess) Close() error {
+	title := "HCSShim::Process::Close"
+
+	if err := hcsCloseProcess(process.handle); err != nil {
```

Comment:
you should set process.handle to 0 once this is done, and check it before calling this function, to avoid double free